lib/tests/streamlit/elements/button_group_test.py

```diff
@@ -106,223 +112,580 @@ def test_deserialize_raise_indexerror(self):
             serde.deserialize([3], "")
 
 
-class ButtonGroupFeedbackTest(DeltaGeneratorTestCase):
-    """Test ability to marshall button_group protos."""
+class TestFeedbackCommand(DeltaGeneratorTestCase):
+    """Tests that are specific for the feedback command."""
 
-    def test_feedback(self):
-        st.feedback("thumbs")
+    @parameterized.expand(
+        [
+            ("thumbs", list(_THUMB_ICONS)),
+            ("faces", list(_FACES_ICONS)),
+            ("stars", list([_STAR_ICON] * 5)),
+        ]
+    )
+    def test_call_feedback_with_all_options(
+        self, option: Literal["thumbs", "faces", "stars"], expected_icons: list[str]
+    ):
+        st.feedback(option)
 
         delta = self.get_delta_from_queue().new_element.button_group
-        correct_thumbs_order = [":material/thumb_up:", ":material/thumb_down:"]
-        self.assertEqual(
-            [option.content for option in delta.options],
-            correct_thumbs_order,
+        assert delta.default == []
+        assert [option.content_icon for option in delta.options] == expected_icons
+
+    def test_invalid_option_literal(self):
+        with pytest.raises(StreamlitAPIException) as e:
+            st.feedback("foo")
+        assert (
+            "The options argument to st.feedback must be one of "
+            "['thumbs', 'faces', 'stars']. The argument passed was 'foo'."
+        ) == str(e.value)
+
+    @parameterized.expand([(0,), (1,)])
+    def test_widget_state_changed_via_session_state(self, session_state_index: int):
+        st.session_state.feedback_command_key = session_state_index
+        val = st.feedback("thumbs", key="feedback_command_key")
+        assert val == session_state_index
+
+
+def get_command_matrix(
+    test_args: list[Any], with_st_feedback: bool = False
+) -> list[tuple[Any]]:
+    """Return a test matrix for the different button group commands and the
+    passed arguments.
+
+    If the test args is a list like [("foo", ("a", "b")), ("bar", ("c", "d"))],
+    this function returns following test matrix:
+    [(st.pills, "foo", ("a", "b")), (_interal_button_group, "bar", ("c", "d"))]
```

Comment:
I'm reading this just now. I wonder if the matrix would actual be 
```
[
        (st.pills, "foo", ("a", "b")),
        (st.pills, "bar", ("c", "d")),
        (_interal_button_group, "foo", ("a", "b")),
        (_interal_button_group, "bar", ("c", "d")),
  ]
```
given the nested loop

Reply:
oh yeah, I have updated the comment! thanks 🙂